.github/actions/yarn-install/action.yml

```diff
@@ -0,0 +1,27 @@
+name: Install Dependencies
+inputs:
+  node-version-type:
+    required: false
+    default: 'current'
+    type: choice
+    options:
+      - 'current'
+      - 'min-supported'
+  min-supported-node-version:
+    required: false
+    default: '18.12'
+  current-node-version:
+    required: false
+    default: '20.2'
+
+runs:
+  using: "composite"
+  steps:
+  - uses: actions/setup-node@v4
+    with:
+      node-version: ${{ inputs.node-version-type == 'current' && inputs.current-node-version || inputs.min-supported-node-version }}
```

Comment:
I'm not sure what value this `node-version-type` indirection gives us - why not pass `node-version` through directly from the matrix, and then we can set it once at the top level?

Reply:
It certainly can be done but I wanted to avoid the different workflows using the action (true there's only one at this point) from passing their own `node-version` to avoid setting it separately in different files.